Detectors/GlobalTrackingWorkflow/tofworkflow/src/CalibInfoReaderSpec.cxx

```diff
@@ -46,30 +47,33 @@ void CalibInfoReader::run(ProcessingContext& pc)
   }
 
   char filename[100];
-  int ientry = 0;
-  while (fscanf(mFile, "%s", filename) == 1) {
-    TFile* fin = TFile::Open(filename);
-    TTree* tin = (TTree*)fin->Get("calibTOF");
-    tin->SetBranchAddress("TOFCalibInfo", &mPvect);
-    for (int i = 0; i < tin->GetEntries(); i += mNinstances) {
-      if ((ientry % mNinstances) == mInstance) {
-        tin->GetEvent(i);
-        LOG(INFO) << "Send " << mVect.size() << " calib infos";
-        pc.outputs().snapshot(Output{o2::header::gDataOriginTOF, "CALIBINFOS", 0, Lifetime::Timeframe}, mVect);
-      }
-      ientry++;
+
+  if ((mTree && mCurrentEntry < mTree->GetEntries()) || fscanf(mFile, "%s", filename) == 1) {
+    if (!mTree || mCurrentEntry >= mTree->GetEntries()) {
+      TFile* fin = TFile::Open(filename);
+      mTree = (TTree*)fin->Get("calibTOF");
+      mCurrentEntry = 0;
+      mTree->SetBranchAddress("TOFCalibInfo", &mPvect);
+    }
+    if ((mGlobalEntry % mNinstances) == mInstance) {
+      mTree->GetEvent(mCurrentEntry);
+      LOG(INFO) << "Send " << mVect.size() << " calib infos";
+      pc.outputs().snapshot(Output{o2::header::gDataOriginTOF, "CALIBDATA", 0, Lifetime::Timeframe}, mVect);
+      usleep(10000);
     }
+    mGlobalEntry++;
+    mCurrentEntry++;
+  } else {
```

Comment:
This part was written by @noferini . The idea, if I am not wrong, is to be able to use the output of several simulation jobs and give these as input to the calibration instead of the DataGenerator device. So one does not want to do a chain, but send in output the content of each file as if it came from different TFs instead of different jobs.

@noferini can comment more.

Reply:
At that time I got some problems with the TChain but I don't remember why. Peraphs the problem was due to something else that I fixed later. I will replace it with a TChain.

Reply:
Maybe we can merge this PR and leave this for another one?